integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/highlight/HighlighterFastVectorIT.java

```diff
@@ -0,0 +1,222 @@
+						"This string mentions a <em>dog</em>"
+				)
+		);
+	}
+
+	protected List<List<String>> orderByScoreResult() {
+		return Arrays.asList(
+				Arrays.asList(
+						"This <em>string</em> mentions a <em>dog</em>",
+						"This <em>string</em> mentions a fox",
+						"jumps right over the little lazy <em>dog</em> This"
+				)
+		);
+	}
+
+	@Test
+	public void boundaryScannerCharsExplicit() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h -> h.highlighterFastVector()
+						.boundaryScanner()
+						.chars()
+						.locale( Locale.ENGLISH )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList( "Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's",
+								"style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>." )
+				);
+	}
+
+	@Test
+	public void boundaryScannerWord() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h -> h.highlighterFastVector()
+						.boundaryScanner()
+						.word()
+						.locale( Locale.ENGLISH )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList(
+								"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's",
+								" style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
+						)
+				);
+	}
+
+	@Test
+	public void boundaryScannerSentence() {
+		assumeTrue( "Ignoring the test on purpose as the results look not ok.", false );
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h -> h.highlighterFastVector()
+						.boundaryScanner()
+						.sentence()
+						.locale( Locale.ENGLISH )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						// this test will fail at lest on ES as it adds one more unexpected sentence to the first match, while I'd expect to have single sentence per line.
+						// produces :
+						//		"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's inception, its musical style has ranged from hard rock, heavy metal and glam metal to soft rock. ",
+						//		"Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>.
```

Comment:
Maybe that's related to the fragment size... ?

Reply:
Trying to guess these highlighter config settings feels like playing minesweeper 😄. If you set the fragment size too low - you might end up having this:
```
"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. ",
    "Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft rock. ",
    "Since the band's inception, its musical style has ranged from hard rock, heavy metal and glam metal to soft <em>rock</em>."
```
But if the value is a bit too high... sentences are combined:
```
"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's inception, its musical style has ranged from hard rock, heavy metal and glam metal to soft rock. ",
    "Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
```